src/main/scala/org/broadinstitute/dsde/workbench/sam/service/TosService.scala

```diff
@@ -27,8 +35,26 @@ class TosService (val directoryDao: DirectoryDAO, val appsDomain: String) extend
     s"tos_accepted_${currentVersion}"
   }
 
-  def getTosGroup(currentVersion: Int): IO[Option[BasicWorkbenchGroup]] = {
-    directoryDao.loadGroup(WorkbenchGroupName(getGroupName(currentVersion)), SamRequestContext(None))
+  def getTosGroup(): IO[Option[BasicWorkbenchGroup]] = {
+    directoryDao.loadGroup(WorkbenchGroupName(getGroupName(tosConfig.version)), SamRequestContext(None))
+  }
+
+  def acceptTosStatus(user: WorkbenchSubject): IO[Boolean] = {
+    if (tosConfig.enabled) {
+      createNewGroupIfNeeded().flatMap {
+        case Some(group) => directoryDao.addGroupMember(group.id, user, SamRequestContext(None))
+        case None => IO.raiseError(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"Terms of Service group ${getGroupName(tosConfig.version)} failed to create.")))
+      }
+    } else IO.pure(false)
+  }
+
+  def getTosStatus(user: WorkbenchSubject): IO[Option[Boolean]] = {
+    if (tosConfig.enabled) {
+      createNewGroupIfNeeded().flatMap {
```

Comment:
Does this mean we'll query the database to see if the Tos group exists each time we check a user's tos agreement status?  Or am I missing some caching?